### PR TITLE
APIO-139 As an APIO API consumer I want the root endpoint to be available in JSON-LD/Hydra

### PR DESCRIPTION
**Pull request by ahdezma**
A Hypermedia API root endpoint (or entry point) is the most important endpoint of all since is the one consumers will know about and from where they will navigate to the different resources the API provides.

Current implementation shows this "entry point" in a format known as "JSON Home" which provides a nice start but doesn't couple well with the rest of the API consumer if, for example, JSON-LD is intended to be used.

The goal of this issue is to generalize the creation of this "root endpoint" so different formats can be used (ideally: "application/json-home", "application/ld+json", "application/hal+json".

### CODE REVIEW

File: apio-architect-impl/src/main/java/com/liferay/apio/architect/impl/message/json/EntryPointMessageMapper.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+ * the provided part of the resource to its representation in a JSON object. To
+ * enable this, each method receives a {@link JSONObjectBuilder}.
+ *
+ * <p>
+ * The method {@link #onFinish} is called when the writer finishes writing the
+ * entry point. Otherwise, the entry point message mapper's methods aren't
+ * called in a particular order.
+ * </p>
+ *
+ * @author Alejandro Hernández
```
Comment (nhpatt) [inline apio-architect-impl/src/main/java/com/liferay/apio/architect/impl/message/json/EntryPointMessageMapper.java]: @review

File: apio-architect-impl/src/main/java/com/liferay/apio/architect/impl/message/json/EntryPointMessageMapper.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+	/**
+	 * Maps a resource URL to its JSON object representation.
+	 *
+	 * @param entryPointJSONObjectBuilder the JSON object builder for the entry
+	 *        point
+	 * @param itemJSONObjectBuilder the JSON object builder for the item
+	 * @param url the resource's URL
+	 */
+	public default void mapItemSelfURL(
```
Comment (nhpatt) [inline apio-architect-impl/src/main/java/com/liferay/apio/architect/impl/message/json/EntryPointMessageMapper.java]: Using self is strange, and what is the difference between mapItem and mapSelf? Why not mapResourceURL and mapEntryPointURL?

File: apio-architect-impl/src/main/java/com/liferay/apio/architect/impl/message/json/EntryPointMessageMapper.java (hunk 1 above)
Comment (ahdezma) [reply apio-architect-impl/src/main/java/com/liferay/apio/architect/impl/message/json/EntryPointMessageMapper.java]: https://github.com/liferay/com-liferay-apio-architect/pull/255/commits/0673374318c3c2232ff18b7c198c81d825259ee0 :)

Comment (nhpatt) [reply apio-architect-impl/src/main/java/com/liferay/apio/architect/impl/message/json/EntryPointMessageMapper.java]: I was going to change it myself...

File: apio-architect-impl/src/main/java/com/liferay/apio/architect/impl/message/json/EntryPointMessageMapper.java (hunk 2 above)
Comment (nhpatt) [reply apio-architect-impl/src/main/java/com/liferay/apio/architect/impl/message/json/EntryPointMessageMapper.java]: But it's ok, I won't leave you hanging.

Comment (ahdezma) [reply apio-architect-impl/src/main/java/com/liferay/apio/architect/impl/message/json/EntryPointMessageMapper.java]: During this months we can make that expected mapper-writer refactoring and pay attention to things like this :)

**Pull request merged**